### DISCUSSION TITLE
how to handle the response from the looping card's OnLoad event? The response get overwritten by the last API call.

### DISCUSSION THREAD

**Discussion by gaurikalp** [EnsembleUI/ensemble#173] [General]
As we call the Slot API on the onLoad event of the doctor card, the slot card is overwritten by the previous API call, 
as recorded in the clip.

https://user-images.githubusercontent.com/115991454/213126524-edda71ca-e0e6-4bad-be07-a47ae0f54de7.mp4

**Comment by vusters** [comment]
@gaurikalp this is an anti-pattern and will have a big performance impact if you list is large. Instead of repeating the API for each card, can you call a single API that returns just the high level data for *all* the cards? That way you can render the cards quickly. Does that work for you?

**Reply by gaurikalp** [reply]
Yes, By calling the Single API with nested JSON, this problem was fixed.
